docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc

```diff
@@ -282,3 +282,30 @@ $JBOSS_HOME/bin/elytron-tool.sh filesystem-realm-encrypt -i ./standalone/configu
 This command creates a new filesystem realm by taking the existing filesystem realm from the specified input location, `./standalone/configuration/fs-realm-plain`, and encrypting its contents using the secret key with alias `key` from the specified credential store, `./mycredstore.cs`. The new filesystem realm is stored in the specified output location, `./standalone/configuration/fs-realm-enc`.
 
 A `.cli` script will also be generated at the root of the filesystem realm. The script contains WildFly CLI commands that can be used to configure a `secret-key-credential-store` resource and a `filesystem-realm` resource in the Elytron subsystem that makes use of the newly encrypted realm content.
+
+== Adding Integrity Checking for an Existing Filesystem Realm
+
+Integrity checking can also be added to an existing filesystem realm using the WildFly Elytron Tool. This takes an existing filesystem realm and creates a new filesystem realm with integrity checking support. The integrity checking uses a set of private-public key pairs, where the private key is only shared with certain known and trusted parties. As a result, the filesystem is secured and someone with access to the identity files, can no longer tamper with the data. 
+
+The feature also creates a CLI script that can be used to add a new filesystem-realm resource to the Elytron subsystem and modify the new filesystem realm with the added integrity checking. 
+
+In order to use this feature, we can make use of the `elytron-tool.sh` (or .bat for windows) script. The format for the command to add integrity checking is as follows:
+```
+$ WILDFLY_HOME/bin/elytron-tool.sh filesystem-realm-integrity --input-location <existing_path> --output-location <new_path> --keystore <ks_path> --password <ks_password> --summary
+```
+
+The required commands include: 
+* input-location: This is the path for the existing filesystem realm that does not have an integrity checking. This must be the path to the directory containing the filesystem, not the path to the filesystem itself. This can also be specified using `-i`. This must be an existing path. 
+* output-location: This attribute will be the path for the new filesystem realm. This filesystem realm will include integrity checking. This can also be specified using `-o`. This path is also a path to a directory.
+* keystore: Path for the keystore that holds the key pair for integrity checking. Can also be specified using `-k`. 
+* password: This is the password for the keystore. This will be specified by the user when creating the keystore. This can also be specified using `-p`. 
```

Comment:
To simplify things a bit I think we can remove "This will be specified by the user when creating the keystore"

Reply:
fixed!